python/cudf/io/json.py

```diff
@@ -8,13 +10,51 @@
 
 
 @ioutils.doc_read_json()
-def read_json(path_or_buf, *args, **kwargs):
-    """{docstring}"""
+def read_json(path_or_buf, engine='cudf', dtype=None, lines=False,
+              compression='infer', byte_range=None, *args, **kwargs):
 
-    warnings.warn("Using CPU via Pandas to read JSON dataset, this may "
-                  "be GPU accelerated in the future")
-    pd_value = pd.read_json(path_or_buf, *args, **kwargs)
-    return cudf.from_pandas(pd_value)
+    """
+    Load and parse a JSON file into a DataFrame
```

Comment:
These parameters only covers the cuDF JSON lines reader, and not the JSON reader that is backed by PyArrow which supports much more.

You can instead update the docstring in `ioutils.doc` to indicate which parameters are supported for cuDF when using cuDF as the engine.

Reply:
Done. The docstring is a bit of a mess now, as many of the parameters differ between engines.